orc8r/cloud/go/services/analytics/calculations/calculations.go

```diff
@@ -40,49 +40,65 @@ const (
 	DirectionLabel = "direction"
 )
 
-//Calculation interface
+// Calculation interface
 type Calculation interface {
 	Calculate(query_api.PrometheusAPI) ([]*protos.CalculationResult, error)
+	GetCalculationParams() *CalculationParams
 }
 
-//MetricConfig is the expected configuration for a specific metric in the config file
+// BaseCalculation Base struct for all calculations
+type BaseCalculation struct {
```

Comment:
I really use BaseCalculation for it to implement a single method GetCalculationParams. Yes it is not neat, but it is a easy way without having to duplicate GetCalculationParams. If you have a cleaner method, let me know. I can use that instead. 

There was a reason why I used pointers. I don't recall it right now. I can change that to using values.

Reply:
I'm just wondering why you're embedding BaseCalculation rather than embedding CalculationParams directly -- perhaps I'm missing some intricacy. But it seems like that would obviate the need for BaseCalculation altogether